### DISCUSSION TITLE
Getting error 403 from VPS server but works on local host

### DISCUSSION THREAD

**Discussion by evauto22** [fastapi/fastapi#11740] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [x] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI 

app = FastAPI()

@app.get("/")  
def root():  
        return ("Success -- the API works!")
```

### Description

I'm creating my first API and running this simple code file titled "test.py" but I can't get it working/deployed on my IONOS VPS as it returns a 403 error (using Postman or browser). FYI it works on my desktop Windows machine via localhost 8000.

The IONOS VPS is running Ubuntu 22.04, Python 3.10.12, and I installed FastAPI and Uvicorn using PIP, and I am using the Plesk control panel SSH Terminal to start the uvicorn server using the command "fastapi run test.py" which appears to return all the successful startup and info messages noted in the FastAPI tutorial doc. However when I try a GET using Postman I get a 403 Forbidden.

I'm not sure this is a valid/relevant test, but I replaced the "test.py" file with a simple "index.html" hello world file and it works when going to the same URL which to me suggests the directory is accessible/available.
  
I'm stumped - any help for a noob appreciated!! 
 
![image](https://github.com/tiangolo/fastapi/assets/155918645/813e12c1-0329-4973-947f-aeb5253f1ba2)
![image](https://github.com/tiangolo/fastapi/assets/155918645/2ef33385-d9bb-4342-ba1e-754a413cd6b5)

### Operating System

Linux

### Operating System Details

Ubuntu 22.04

### FastAPI Version

0.111.0

### Pydantic Version

2.7.4

### Python Version

3.10.12

**Comment by binbjz** [comment]
To solve the 403 Forbidden error on your IONOS VPS, follow these steps:

### 1. Check Plesk Firewall Settings
1. **Login to Plesk Control Panel**.
2. Go to **"Tools & Settings"**.
3. Click on **"Firewall"**.
4. Ensure the port (e.g., 8000) used by your FastAPI server is allowed for external access. Add a rule if necessary.

### 2. Verify Uvicorn is Correctly Started
Make sure Uvicorn is binding to 0.0.0.0 and running on the correct port (8000).
```bash
uvicorn test:app --host 0.0.0.0 --port 8000
```

### 3. Ensure Firewall Allows Port 8000
Check the firewall status and allow port 8000 if it's not already allowed.
```bash
sudo ufw status
sudo ufw allow 8000
```

### 4. Start Uvicorn via Plesk Terminal
1. **Open Plesk Terminal**:
   - Go to **"Tools & Settings"**.
   - Click on **"Web Terminal"** or **"SSH Terminal"**.
2. **Start Uvicorn**:
   ```bash
   uvicorn test:app --host 0.0.0.0 --port 8000
   ```

### 5. Test API Access
Try accessing your API externally using curl or Postman.
```bash
curl --request GET --url http://your_vps_ip:8000/ --header 'Content-Type: application/json'
```

### 6. Optional: Configure Reverse Proxy (Nginx Example)
If using Nginx as a reverse proxy, make sure your configuration is correct.
```nginx
server {
    listen 80;
    server_name your_domain_or_ip;

    location / {
        proxy_pass http://127.0.0.1:8000;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection 'upgrade';
        proxy_set_header Host $host;
        proxy_cache_bypass $http_upgrade;
    }
}
```

By following these steps, you should be able to resolve the 403 Forbidden error and access your FastAPI application on your IONOS VPS.

**Reply by evauto22** [reply]
Thank you for the detailed instructions! Unfortunately, I still can't find success.   I turned off the Firewall (and banned IP addresses) in Plesk.  Uvicorn seems to start correctly giving the message “Uvicorn  running on http://0.0.0.0:8000”.
The command “sudo ufw status” returns “Status:inactive”, and “sudo ufw allow 8000” initially responded with “rules updated” but no change. 

I don't know if this datapoint is helpful/relevant, but if via browser I enter http:app.domain.com/test.py the browser returned the text of the test.py file.  So it seems it is accessible, not sure why the 403.
